src/plugins/index_pattern_management/public/components/field_editor/field_editor.tsx

```diff
@@ -413,6 +418,33 @@ export class FieldEditor extends PureComponent<FieldEdiorProps, FieldEditorState
     );
   }
 
+  renderCustomLabel() {
+    const { customLabel, spec } = this.state;
+
+    return (
+      <EuiFormRow
+        label={i18n.translate('indexPatternManagement.customLabelLabel', {
+          defaultMessage: 'Custom label',
+        })}
+        helpText={
+          <FormattedMessage
+            id="indexPatternManagement.customLabelHelpText"
+            defaultMessage="A custom label is displayed instead of the original field name. It is useful e.g. when the original field name is very long."
+          />
+        }
+      >
+        <EuiFieldText
+          value={customLabel || ''}
+          placeholder={spec.name}
+          data-test-subj="editorFieldCustomLabel"
+          onChange={(e) => {
+            this.setState({ customLabel: e.target.value });
+          }}
+        />
+      </EuiFormRow>
```

Comment:
@andreadelrio I've added help text 
<img width="426" alt="Bildschirmfoto 2020-08-31 um 13 59 07" src="https://user-images.githubusercontent.com/463851/91718723-3f6cbf00-eb94-11ea-8a1c-7c2d713a0af6.png">
FYI @gchaps for text check and suggestions